BEPs/BEP-404.md

```diff
@@ -0,0 +1,55 @@
+<pre>
+  BEP: 404
+  Title: Clear Miner History when Switching Validators Set
+  Status: Candidate
+  Type: Standards
+  Created: 2024-07-01
+</pre>
+
+# BEP-404: Clear Miner History when Switching Validators Set
+
+- [BEP-404: Clear Miner History when Switching Validators Set](#bep-404-clear-miner-history-when-switching-validators-set)
+  - [1. Summary](#1-summary)
+  - [2. Motivation](#2-motivation)
+  - [3. Status](#3-status)
+  - [4. Specification](#4-specification)
+  - [5. Backwards Compatibility](#5-backwards-compatibility)
+  - [6. License](#6-license)
+
+## 1. Summary
+After switching the validator set on the BSC chain, the previous block production record is still used to restrict some validators from producing blocks for the first few blocks after the switch, leading to brief network instability. This BEP aims to resolve this issue.
+
+## 2. Motivation
+The validator set on the BSC chain switches at each epoch.
+
+Assume the number of validators in an epoch is validatorN. To ensure the consistency of the BSC chain, it is required that the chain must have at least validatorN/2+1 validators to continue advancing the chain.
+
+To achieve this constraint, a set called MinerHistory is defined, with a length of validatorN/2, recording the validators that produced the last validatorN/2 blocks. New block-producing validators are required to not be in MinerHistory.
+
+This BEP aims to enhance network stability by clearing MinerHistory when the validator set switches.
+
+## 3. Status
+This BEP is in progress.
+
+## 4. Specification
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+In the previous specification, when the validator set switches at each epoch, MinerHistory is not cleared, causing some validators to be unable to produce blocks normally for the first few blocks after the switch.
+<div align="center">
+<img src=./assets/BEP-404/3-1.png width=100% />
+</div>
+This BEP clarifies that MinerHistory should be cleared when switching the validator set to solve this issue.
+
+```Go
+MinerHistory  map[uint64]Address  
+
+if switching validator set for new epoch {
```

Comment:
1.pls be more specific when will this clear MinerHistory be executed?
Is it after processing a block with block number satisfy: 
```
BlockNum % EpochLength == (ValidatorSize/2)
```
2.Could you add a part about how to handle reorg?

Reply:
1. first suggestion accepted
2. no change about the reorg logic